### PR TITLE
Diff Table Unit Tests

### PR DESCRIPTION
**Pull request by Matt-Cumber**
**Description of work.**
Added unit tests for the main difference table functionality. The functionality is shared between the group differences table and the pair differences table, which is why there is only one set of tests.

bug in file pairing_table_widget_presenter is fixed by #30702

**To test:**
All tests should pass
Anything I missed or that doesn't look right?

part of #30558

*This does not require release notes* because **it is only unit test changes**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/ADSHandler/workspace_naming_test.py [hunk 1]
```diff
@@ -4,12 +4,88 @@
...
+from mantid import ConfigService
+from Muon.GUI.Common.utilities.load_utils import load_workspace_from_filename
+
 
 class WorkspaceNamingTest(unittest.TestCase):
 
+    def setUp(self):
+        return
+
+    def test_getGroupOrPairName(self):
```
Comment (AnthonyLim23) [inline scripts/test/Muon/ADSHandler/workspace_naming_test.py:21]: lets add a diff too

**Comment by AnthonyLim23** [comment]
you could save yourself some work. 

difference table widget presenter should check that adding/removing diffs etc works (doesn't matter if you pick group or pair for the tests)
difference_widget_presenter should just test the enable/disable and subscribers

File: scripts/Muon/GUI/Common/difference_table_widget/difference_table_widget_view.py [hunk 2]
```diff
@@ -307,6 +307,7 @@ def enable_editing(self):
         self.enable_updates()
 
     def disable_editing(self):
+        print("disabling")
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/difference_table_widget/difference_table_widget_view.py:310]:
```suggestion
```

Comment (Matt-Cumber) [reply scripts/Muon/GUI/Common/difference_table_widget/difference_table_widget_view.py:310]: Oops, thanks

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**